examples/src/main/java/ai/djl/examples/training/transferlearning/TransferFreshFruit.java

```diff
@@ -0,0 +1,196 @@
+        config.optOptimizer(optimizer);
+
+        // Config trainer
+        Trainer trainer = model.newTrainer(config);
+        trainer.setMetrics(new Metrics());
+
+        // Initialize the parameter shape and value
+        int batchSize = 32;
+        Shape inputShape = new Shape(batchSize, 3, 224, 224);
+        trainer.initialize(inputShape);
+
+        // Data
+        ImageFolder datasetTrain = getData("test", "banana", batchSize);
+
+        // Train
+        EasyTrain.fit(trainer, 6, datasetTrain, null);
+
+        // Save model
+        // model.save("your-model-path");
+
+        model.close();
+        embedding.close();
+        return null;
+    }
+
+    private static ImageFolder getData(String subfolderName, String fruit, int batchSize)
+            throws TranslateException, IOException {
+        // The dataset is from <a
+        // href="https://www.kaggle.com/datasets/sriramr/fruits-fresh-and-rotten-for-classification">https://www.kaggle.com/datasets/sriramr/fruits-fresh-and-rotten-for-classification</a>
+        String folderUrl = "/Users/fenkexin/Desktop/transferDJL/code/data/" + fruit;
+        String subfolder = "/" + subfolderName + "/";
+        Repository repository = Repository.newInstance("banana", Paths.get(folderUrl + subfolder));
+        ImageFolder dataset =
+                ImageFolder.builder()
+                        .setRepository(repository)
+                        .addTransform(new ToTensor())
+                        .addTransform(new Transpose(1, 2, 0))
+                        .addTransform(new Resize(224, 224))
+                        .addTransform(new Transpose(2, 0, 1))
+                        .addTargetTransform(new OneHot(2))
+                        .setSampling(batchSize, true)
+                        .build();
+        dataset.prepare();
+        return dataset;
+    }
+
+    private static DefaultTrainingConfig setupTrainingConfig(Arguments arguments) {
+        String outputDir = arguments.getOutputDir();
+        SaveModelTrainingListener listener = new SaveModelTrainingListener(outputDir);
+        listener.setSaveModelCallback(
+                trainer -> {
+                    TrainingResult result = trainer.getTrainingResult();
+                    Model model = trainer.getModel();
+                    float accuracy = result.getValidateEvaluation("Accuracy");
+                    model.setProperty("Accuracy", String.format("%.5f", accuracy));
+                    model.setProperty("Loss", String.format("%.5f", result.getValidateLoss()));
+                });
+
+        return new DefaultTrainingConfig(new SoftmaxCrossEntropy("SoftmaxCrossEntropy"))
+                .addEvaluator(new Accuracy())
+                .optDevices(Engine.getInstance().getDevices(1))
+                .addTrainingListeners(TrainingListener.Defaults.logging(outputDir))
+                .addTrainingListeners(listener);
+    }
+
+    private static class SoftmaxCrossEntropy extends Loss {
```

Comment:
Why not use the standard `SoftmaxCrossEntropyLoss`?

Reply:
The standard `SoftmaxCrossEntropyLoss` doesn't take the output of the softmax function; it currently either take the values before applying softmax, or the values after applying `logit = log(softmax)`.
Here, I built the network to output softmax. This is mainly because it is natural to understand (as probabilities). Correspondingly, I also corrected the calculation in Accuracy.java. You can take a look if it is correct.